model/jpa/src/main/java/org/keycloak/connections/jpa/DefaultJpaConnectionProviderFactory.java

```diff
@@ -105,13 +105,16 @@ private EntityManager createEntityManager(KeycloakSession session) {
     }
 
     private void addSpecificNamedQueries(KeycloakSession session, Connection connection) {
-        LiquibaseConnectionProvider liquibaseProvider = session.getProvider(LiquibaseConnectionProvider.class);
         EntityManager em = null;
         try {
-            Liquibase liquibase = liquibaseProvider.getLiquibase(connection, this.getSchema());
             em = createEntityManager(session);
-            JpaUtils.addSpecificNamedQueries(em, liquibase.getDatabase().getShortName());
-        } catch (LiquibaseException e) {
+            String dbKind = connection.getMetaData().getDatabaseProductName();
+            for (Map.Entry<Object, Object> query : loadSpecificNamedQueries(dbKind.toLowerCase()).entrySet()) {
```

Comment:
I think that the `getDatabaseProductName` returns long names for mssql. Checking the liquibase code:

* H2Database: H2
* MySQLDatabase: MySQL
* MariaDBDatabase: MariaDB
* PostgresDatabase: PostgreSQL
* MSSQLDatabase: Microsoft SQL Server or SQLOLEDB
* OracleDatabase: Oracle
* PostgresPlusDatabase: EnterpriseDB

So I think we can better add a little method in the JpaUtils that performs the transformation. It would only affect MSSQL and enterprise postgresql:

```java
    public static String getDatabaseType(String productName) {
        switch (productName) {
            case "Microsoft SQL Server":
            case "SQLOLEDB":
                return "mssql";
            case "EnterpriseDB":
                return "postgresql";
            default:
                return productName.toLowerCase();
        }
    }
```